routes/auth-routes.js

```diff
@@ -1,140 +0,0 @@
-// The routes file will be defined in the routes/auth.routes.js, and we will set the necessary packages and code to signup in the application:
-
-const { Router } = require("express");
-const router = new Router();
-const passport = require("passport");
-const User = require("../models/user");
-const bcrypt = require("bcrypt");
-const bcryptSalt = 10;
-const { isLoggedIn, isLoggedOut } = require("../middleware/route-guard");
-const filteredDomains = require("../public/javascripts/filteredDomains");
-const nodemailer = require("nodemailer");
-const nodemailerSetup = require('../public/javascripts/nodemailer');
-
-// SIGNUP ROUTES, implemented with server-side validation on the email and password used. Users are also issued a verification code over email to verify their sign up.
-
-router.get("/signup", isLoggedOut, (req, res, next) => res.render("auth/signup", { layout: false }));
-
-router.post("/signup", isLoggedOut, (req, res, next) => {
-  const { firstName, lastName, email, password, companyName } = req.body;
-  const regex = /(?=.*\d)(?=.*[a-z])(?=.*[A-Z]).{6,}/;
-  if (!regex.test(password)) {
-    res.status(500).render("auth/signup", { layout: false, errorMessage: "Password needs to have at least 6 chars and must contain at least one number, one lowercase and one uppercase letter." });
-    return;
-  }
-  if (filteredDomains.indexOf(email.split("@")[1]) !== -1) {
-    res.status(500).render("auth/signup", { layout: false, errorMessage: "Email needs to be a professional email address." });
-    return;
-  }
-  if (!email || !password) {
-    console.log("USERNAME OR PASSWORD MISSING");
-    res.render("auth/signup", {
-      layout: false,
-      errorMessage: "Indicate username and password",
-    });
-    return;
-  }
-  const characters = "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
```

Comment:
This confirmation code is not directly related to the user signup route so I think it's a good idea to bring this out to a util function

Reply:
I'd also add a comment explaining why we have it